### DISCUSSION TITLE
Bidirectional relationships and more control over relationships in relational DBs

### DISCUSSION THREAD

**Discussion by jmikrut** [payloadcms/payload#4300] [Roadmap]
Right now, in Postgres, all relationships for a given document are stored in a single join table, which is for performance and simplicity reasons. This way, any number of relationship fields across a document, at any level or within any sub-field structure, are stored and retrieved in a single join.

But for relations that are `hasMany: false` and not polymorphic (i.e. have a single `relationTo`), you might want to have a more expected database structure where rather than storing a relation in the existing join table, you instead just store the related ID directly on the row in the relevant table.

We should expose a set of options to allow for more fine-grained control here.

There might be some other cases where we can add flexibility. Let's discuss it here.

EDIT: 

**Bidirectional Relationships** are also planned.

Related:
https://github.com/payloadcms/payload/discussions/4163

EDIT:
**RELEASED IN [3.0](https://github.com/payloadcms/payload/releases/tag/v3.0.0)**

**Comment by paulpopus** [comment]
Note on this, some people have asked about supporting junction tables as well for many-to-many relationships

**Reply by abass** [reply]
Yes! This please 🙏

**Comment by jesse996** [comment]
any progress?

**Reply by abass** [reply]
Yes, this is the main thing preventing us currently as well, any update on ETA/Status update would be super appreciated 🙏

**Comment by nachomglz** [comment]
any updates on this? it's been really a pain to work around this having to create custom sql queries.

**Reply by andershermansen** [reply]
Join fields are available in latest 3.0 beta. Check the documentation here: https://payloadcms.com/docs/beta/fields/join
